libp2p/network/url_conversion_test.go

```diff
@@ -63,56 +63,73 @@ func TestPeerMultiAddrFromServiceURL_Error(t *testing.T) {
 	hostnames := map[string]string{
 		fqdn: "www.google.com",
 		ip4:  "142.250.181.196",
-		ip6:  "2a00:1450:4005:802::2004",
+		ip6:  "[2a00:1450:4005:802::2004]",
 	}
 
 	testCases := []struct {
 		name             string
 		serviceURLFormat string
-		// TECHDEBT: assert specific errors.
-		expectedErrContains string
+		expectedErr      string
 	}{
 		// Usage of scheme is invalid.
 		{
 			"fully qualified domain name with scheme",
 			"tcp://%s:8080",
-			"resolving peer IP for hostname",
+			errInvalidSchemeUsageMsg,
 		},
 
 		// Port **number** is required
 		{
 			"invalid port number",
 			"%s:abc",
-			"invalid port",
+			errParsingServiceURLMsg,
 		},
 		{
 			"missing port number",
 			"%s:",
-			"unexpected end of multiaddr",
+			errMissingPortAndDelimiterMsg,
+		},
+		{
+			"missing port number and delimiter",
+			"%s",
+			errMissingPortAndDelimiterMsg,
 		},
-		// TODO: this case is tricky to detect as IPv6 addresses
-		// can omit multiple "hextet" delimiters and still be valid.
-		// (see: https://en.wikipedia.org/wiki/IPv6#Address_representation)
-		// {
-		// 	"missing port number and delimiter",
-		// 	"%s",
-		// },
 	}
 
 	for _, testCase := range testCases {
 		for hostType, hostname := range hostnames {
 			testName := fmt.Sprintf("%s/%s", testCase.name, hostType)
 			t.Run(testName, func(t *testing.T) {
 				serviceURL := fmt.Sprintf(testCase.serviceURLFormat, hostname)
-				actualMultiaddr, err := Libp2pMultiaddrFromServiceURL(serviceURL)
-				// TECHDEBT: assert specific errors
-				// Print resulting multiaddr to understand why no error.
-				require.ErrorContainsf(t, err, testCase.expectedErrContains, fmt.Sprintf("actualMultiaddr: %s", actualMultiaddr))
+				_, err := Libp2pMultiaddrFromServiceURL(serviceURL)
+				require.ErrorContainsf(t, err, testCase.expectedErr, fmt.Sprintf("expected error to start with %q, but got %q", testCase.expectedErr, err.Error()))
```

Comment:
Much better, thank you! :raised_hands:

It looks like the `msg` arg to `ErrContainsf` is redundant with respect to what the test framework prints when the test fails. I think we can reduce this message to something like `"error message doesn't match"`. My apologies for not articulating the specific ask more clearly the first time around. :sweat_smile: 

This is the output when it fails, as currently written (I forced failure by replacing the expected error message):
```
    url_conversion_test.go:105: 
        	Error Trace:	/home/bwhite/Projects/pokt/pocket/libp2p/network/url_conversion_test.go:105
        	Error:      	Error "missing port number and delimiter in service URL: scheme://142.250.181.196: address 142.250.181.196: missing port in address" does not contain "<expected error message goes here - FORCED FAILURE>"
        	Test:       	TestPeerMultiAddrFromServiceURL_Error/missing_port_number_and_delimiter/ip4
        	Messages:   	expected error to start with "<expected error message goes here - FORCED FAILURE>", but got "missing port number and delimiter in service URL: scheme://142.250.181.196: address 142.250.181.196: missing port in address"
```

This is what it would look like if we simplify the `msg` argument:
```
=== RUN   TestPeerMultiAddrFromServiceURL_Error/missing_port_number_and_delimiter/fqdn
    url_conversion_test.go:106: 
        	Error Trace:	/home/bwhite/Projects/pokt/pocket/libp2p/network/url_conversion_test.go:106
        	Error:      	Error "missing port number and delimiter in service URL: scheme://www.google.com: address www.google.com: missing port in address" does not contain "<expected error message goes here - FORCED FAILURE>"
        	Test:       	TestPeerMultiAddrFromServiceURL_Error/missing_port_number_and_delimiter/fqdn
        	Messages:   	error message doesn't match
    --- FAIL: TestPeerMultiAddrFromServiceURL_Error/missing_port_number_and_delimiter/fqdn (0.00s)
```

Reply:
ah great suggestion, i simplified it to `unexpected error`.